builder/dockerfile/builder.go

```diff
@@ -130,24 +129,28 @@ type Builder struct {
 }
 
 // newBuilder creates a new Dockerfile builder from an optional dockerfile and a Options.
-func newBuilder(clientCtx context.Context, options builderOptions) (*Builder, error) {
+func newBuilder(ctx context.Context, options builderOptions) (*Builder, error) {
 	config := options.Options
 	if config == nil {
 		config = new(types.ImageBuildOptions)
 	}
 
+	imageProber, err := newImageProber(ctx, options.Backend, config.CacheFrom, config.NoCache)
+	if err != nil {
+		return nil, err
+	}
+
 	b := &Builder{
-		clientCtx:        clientCtx,
 		options:          config,
 		Stdout:           options.ProgressWriter.StdoutFormatter,
 		Stderr:           options.ProgressWriter.StderrFormatter,
 		Aux:              options.ProgressWriter.AuxFormatter,
 		Output:           options.ProgressWriter.Output,
 		docker:           options.Backend,
 		idMapping:        options.IDMapping,
-		imageSources:     newImageSources(clientCtx, options),
+		imageSources:     newImageSources(ctx, options),
```

Comment:
A function call with a context argument but no error return value sets my spidey-senses tingling. Looking at the implementation, it's worse than I feared. On the one hand, it's been like that for five years, but on the other hand it is directly related to plumbing contexts into ImageService. I leave it up to you to decide whether to address it in this PR or in a follow-up.

Reply:
Thanks! Moved the context to `Get` function instead of sneakily capturing it via closure.